File: suivi-scolaire-projet-complet/backend-laravel/app/Models/Matiere.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Matiere extends Model
{
    protected $fillable = [
        'nom',
        'coefficient',
        'note_sur'
    ];

    public function classes(){
    return $this->belongsToMany(Classe::class, 'classe_matiere');
    }

    #Méthode de notes. Dans chaque matière ont peut avoir plusieurs notes
    public function notes(){
        return $this->hasMany(Note::class);
    } 
}

File: suivi-scolaire-projet-complet/mobile-android/app/src/main/java/com/ecole/suiviscolaire/data/model/Matiere.kt
package com.ecole.suiviscolaire.data.model

import com.google.gson.annotations.SerializedName

data class Matiere(
    @SerializedName("id") val id: Int,
    @SerializedName("nom") val nom: String,
    @SerializedName("coefficient") val coefficient: Int,
    @SerializedName("note_sur") val noteSur: Int? = null
)
